### ISSUE TITLE
API doc table not rendering for fold_gates_at_random

### ISSUE THREAD

**Issue by crazy4pi314** [unitaryfund/mitiq#503]
Pre-Report Checklist
--------------------

- [x] I am running the latest version of mitiq
- [x] I checked to make sure that this bug has not already been reported

Issue Description
-----------------

The table in this API docstring doesn't render: 

https://mitiq.readthedocs.io/en/stable/apidoc.html?highlight=fold%20random#mitiq.zne.scaling.folding.fold_gates_at_random 

How to Reproduce
----------------

Not sure from sphinx how to get it to parse this as a markdown table, I get the following sphinx message when building locally: 

### Code Snippet

```bash
make docs
```

### Error Output

```
/workspaces/mitiq/mitiq/zne/scaling/folding.py:docstring of mitiq.zne.scaling.folding.fold_gates_at_random:26: WARNING: Unexpected section title.

Gate key    | Gate
-------------------------
/workspaces/mitiq/mitiq/zne/scaling/folding.py:docstring of mitiq.zne.scaling.folding.fold_gates_from_left:25: WARNING: Unexpected section title.

Gate key    | Gate
-------------------------
/workspaces/mitiq/mitiq/zne/scaling/folding.py:docstring of mitiq.zne.scaling.folding.fold_gates_from_right:25: WARNING: Unexpected section title.

Gate key    | Gate
-------------------------
```

Environment Context
-------------------

Use the `about()` function to summarize information on operating system, python version and dependencies.
```python
from mitiq import about
about()

Mitiq: A Python toolkit for implementing error mitigation on quantum computers
==============================================================================
Authored by: Mitiq team, 2020 & later (https://github.com/unitaryfund/mitiq)

Mitiq Version:  0.5.0dev

Core Dependencies
-----------------
Cirq Version:   0.9.1
NumPy Version:  1.18.5
SciPy Version:  1.6.0

Optional Dependencies
---------------------
PyQuil Version: 2.18.0
Qiskit Version: 0.12.0

Python Version: 3.8.6
Platform Info:  Linux (x86_64)
```

Additional Python Environment Details (`pip freeze` or `conda list`):

```
absl-py==0.11.0
alabaster==0.7.12
antlr4-python3-runtime==4.7.2
appdirs==1.4.4
argon2-cffi==20.1.0
arrow==0.17.0
astunparse==1.6.3
async-generator==1.10
attrs==20.3.0
Babel==2.9.0
backcall==0.2.0
black==19.10b0
bleach==3.2.2
cachetools==4.2.0
certifi==2020.12.5
cffi==1.14.4
chardet==4.0.0
cirq==0.9.1
click==7.1.2
colorama==0.4.4
coverage==5.3.1
cryptography==3.3.1
cycler==0.10.0
Cython==0.29.21
decorator==4.4.2
defusedxml==0.6.0
dill==0.3.3
dlx==1.0.4
docplex==2.19.202
docutils==0.16
entrypoints==0.3
fastdtw==0.3.4
flake8==3.7.9
freezegun==0.3.15
gast==0.3.3
gitdb==4.0.5
GitPython==3.1.12
google-api-core==1.25.0
google-auth==1.24.0
google-auth-oauthlib==0.4.2
google-pasta==0.2.0
googleapis-common-protos==1.52.0
grpcio==1.35.0
h5py==2.10.0
idna==2.10
imagesize==1.2.0
immutables==0.6
importlib-metadata==3.4.0
inflection==0.5.1
ipykernel==5.4.3
ipython==7.19.0
ipython-genutils==0.2.0
ipywidgets==7.6.3
jedi==0.18.0
Jinja2==2.11.2
joblib==1.0.0
jsonschema==3.2.0
jupyter-cache==0.4.2
jupyter-client==6.1.11
jupyter-core==4.7.0
jupyter-sphinx==0.3.1
jupyterlab-pygments==0.1.2
jupyterlab-widgets==1.0.0
Keras-Preprocessing==1.1.2
kiwisolver==1.3.1
latexcodec==2.0.1
m2r==0.2.1
Markdown==3.3.3
markdown-it-py==0.6.1
MarkupSafe==1.1.1
marshmallow==3.10.0
marshmallow-polyfield==5.9
matplotlib==3.2.2
mccabe==0.6.1
mdit-py-plugins==0.2.4
mistune==0.8.4
more-itertools==8.6.0
mpmath==1.1.0
msgpack==0.6.2
mypy==0.790
mypy-extensions==0.4.3
myst-nb==0.11.1
myst-parser==0.13.3
nbclient==0.5.1
nbconvert==5.6.1
nbdime==2.1.0
nbformat==5.1.2
nest-asyncio==1.4.3
networkx==2.5
notebook==6.1.6
ntlm-auth==1.5.0
numpy==1.18.5
oauthlib==3.1.0
opt-einsum==3.3.0
oset==0.1.3
packaging==20.8
pandas==1.2.1
pandocfilters==1.4.3
parso==0.8.1
pathspec==0.8.1
pexpect==4.8.0
pickleshare==0.7.5
pluggy==0.13.1
ply==3.11
prometheus-client==0.9.0
prompt-toolkit==3.0.11
protobuf==3.14.0
psutil==5.8.0
ptyprocess==0.7.0
py==1.10.0
pyasn1==0.4.8
pyasn1-modules==0.2.8
pybind11==2.6.1
pybtex==0.22.2
pybtex-docutils==1.0.0
pycodestyle==2.5.0
pycparser==2.20
pydata-sphinx-theme==0.3.2
pyflakes==2.1.1
Pygments==2.7.4
pyparsing==2.4.7
pyquil==2.18.0
pyrsistent==0.17.3
pyscf==1.7.5.1
pytest==5.4.3
pytest-cov==2.8.1
python-dateutil==2.8.1
python-rapidjson==1.0
pytz==2020.5
PyYAML==5.4
pyzmq==21.0.1
qiskit==0.16.2
qiskit-aer==0.4.1
qiskit-aqua==0.6.5
qiskit-ibmq-provider==0.5.0
qiskit-ignis==0.2.0
qiskit-terra==0.12.0
Quandl==3.6.0
regex==2020.11.13
requests==2.25.1
requests-ntlm==1.1.0
requests-oauthlib==1.3.0
rpcq==3.7.0
rsa==4.7
ruamel.yaml==0.16.12
ruamel.yaml.clib==0.2.2
scikit-learn==0.24.1
scipy==1.6.0
Send2Trash==1.5.0
six==1.15.0
smmap==3.0.4
snowballstemmer==2.0.0
sortedcontainers==2.3.0
Sphinx==2.4.4
sphinx-autodoc-typehints==1.10.3
sphinx-copybutton==0.2.12
sphinx-togglebutton==0.2.3
sphinxcontrib-applehelp==1.0.2
sphinxcontrib-bibtex==1.0.0
sphinxcontrib-devhelp==1.0.2
sphinxcontrib-htmlhelp==1.0.3
sphinxcontrib-jsmath==1.0.1
sphinxcontrib-qthelp==1.0.3
sphinxcontrib-serializinghtml==1.1.4
SQLAlchemy==1.3.22
sympy==1.5
tensorboard==2.4.1
tensorboard-plugin-wit==1.8.0
tensorflow==2.3.2
tensorflow-estimator==2.3.0
tensorflow-quantum==0.4.0
termcolor==1.1.0
terminado==0.9.2
testpath==0.4.4
threadpoolctl==2.1.0
toml==0.10.2
tornado==6.1
traitlets==5.0.5
typed-ast==1.4.2
typing-extensions==3.7.4.3
urllib3==1.26.2
wcwidth==0.2.5
webencodings==0.5.1
websockets==7.0
Werkzeug==1.0.1
widgetsnbextension==3.5.1
wrapt==1.12.1
zipp==3.4.0
```

**Comment by rmlarose** [comment]
Ah, good find Sarah. Probably can be fixed by a raw string `r"""..."""`, if not just formatting the table as markdown. Also note the same table appears in other folding function docstrings (see error output above).

**Comment by crazy4pi314** [comment]
This is fixed now, and will be merged with the mystdocs PR #509.
